src/org/zaproxy/zap/extension/ascanrulesAlpha/XSLTInjection.java

```diff
@@ -0,0 +1,176 @@
+import org.zaproxy.zap.model.Vulnerabilities;
+import org.zaproxy.zap.model.Vulnerability;
+
+/**
+ * The XSLTInjection plugin identifies XSLT injection vulnerabilities with all the parameters in GET and POST
+ *
+ * @author CaptainFreak
+ */
+public class XSLTInjection extends AbstractAppParamPlugin {
+    
+	// plugin dependencies
+    private static final String[] dependency = {};
+    
+	//For logging
+	private static final Logger LOG = Logger.getLogger(XSLTInjection.class);
+
+    private static final String[] errorCausingPayloads={"<",">","\'","\""};
+    
+    private static final String[] vendorReturningPayloads={"<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/>",
+    														"system-property(\'xsl:vendor\')/>",
+    														"\"/><xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--",
+    														"<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--"
+    														};
+    private static final String[] xsltVendors={"libxslt","Microsoft","Saxonica","Apache","Xalan","SAXON","Transformiix"};
+    
+    //TODO Add time based payloads (SSRF)
+    
+	@Override
+	public int getId() {
+		return 90017;
+	}
+
+	@Override
+	public String getName() {		
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.name");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return dependency;
+	}
+
+	@Override
+	public boolean targets(TechSet technologies) {  
+		return technologies.includes(Tech.XML); 
+	}
+	
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.refs");
+	}
+
+	@Override
+	public void init() {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Initialising XSLT Scanner.");
+        }
+	}
+
+	/*
+	 * This method is called by the active scanner for each GET and POST parameter for every page 
+	 * @see org.parosproxy.paros.core.scanner.AbstractAppParamPlugin#scan(org.parosproxy.paros.network.HttpMessage, java.lang.String, java.lang.String)
+	 */
+	
+	@Override
+	public void scan(HttpMessage msg, String param, String value) {
+		try {
+			//Count positive Tests
+			int positiveBasedOnError=0; 
+			//error causing tests
+			for (int i=0;i<errorCausingPayloads.length;i++){
+				msg = getNewMsg();
+				setParameter(msg, param, errorCausingPayloads[i]);
+				sendAndReceive(msg);
+				if(!msg.getResponseBody().toString().contains(errorCausingPayloads[i])){
+					positiveBasedOnError++;
+				}
+				// Stop the testing if user requests
+				if (isStop()) {
+					if (LOG.isDebugEnabled())
+						LOG.debug("Stopping the scan due to a user request");
```

Comment:
While this is syntactically correct it would be nice if all the if blocks consistently used curly braces. (Throughout)